src/allmydata/immutable/upload.py

```diff
@@ -553,6 +592,7 @@ def _got_response(self, res, tracker, shares_to_ask, put_tracker_here):
                     level=log.NOISY)
             progress = False
             for s in alreadygot:
+                self.peer_selector.confirm_share_allocation(s, tracker.get_serverid())
```

Comment:
This appears to do the inverse of Step 10, but I haven't located Step 10 itself yet.